apps/bfd-db-migrator/src/main/java/gov/cms/bfd/migrator/app/AppConfiguration.java

```diff
@@ -0,0 +1,334 @@
+import java.util.function.Function;
+
+/** Models the configuration options for the application. */
+public final class AppConfiguration implements Serializable {
+  private static final long serialVersionUID = -6845504165285244536L;
+
+  /**
+   * The name of the environment variable that should be used to provide the {@link
+   * #getDatabaseOptions()} {@link DatabaseOptions#getDatabaseUrl()} value.
+   */
+  public static final String ENV_VAR_KEY_DATABASE_URL = "DATABASE_URL";
+
+  /**
+   * The name of the environment variable that should be used to provide the {@link
+   * #getDatabaseOptions()} {@link DatabaseOptions#getDatabaseUsername()} value.
+   */
+  public static final String ENV_VAR_KEY_DATABASE_USERNAME = "DATABASE_USERNAME";
+
+  /**
+   * The name of the environment variable that should be used to provide the {@link
+   * #getDatabaseOptions()} {@link DatabaseOptions#getDatabasePassword()} value.
+   */
+  public static final String ENV_VAR_KEY_DATABASE_PASSWORD = "DATABASE_PASSWORD";
+
+  /**
+   * The name of the environment variable that should be used to provide the {@link
+   * #getDatabaseOptions()} {@link DatabaseOptions#getMaxPoolSize()} value.
+   */
+  public static final String ENV_VAR_KEY_DATABASE_MAX_POOL_SIZE = "DATABASE_MAX_POOL_SIZE";
+
+  /**
+   * The name of the environment variable that should be used to provide the {@link
+   * #getMetricOptions()} {@link MetricOptions#getNewRelicMetricKey()} value.
+   */
+  public static final String ENV_VAR_NEW_RELIC_METRIC_KEY = "NEW_RELIC_METRIC_KEY";
+
+  /**
+   * The name of the environment variable that should be used to provide the {@link
+   * #getMetricOptions()} {@link MetricOptions#getNewRelicAppName()} value.
+   */
+  public static final String ENV_VAR_NEW_RELIC_APP_NAME = "NEW_RELIC_APP_NAME";
+
+  /**
+   * The name of the environment variable that should be used to provide the {@link
+   * #getMetricOptions()} {@link MetricOptions#getNewRelicMetricHost()} value.
+   */
+  public static final String ENV_VAR_NEW_RELIC_METRIC_HOST = "NEW_RELIC_METRIC_HOST";
+
+  /**
+   * The name of the environment variable that should be used to provide the {@link
+   * #getMetricOptions()} {@link MetricOptions#getNewRelicMetricPath()} value.
+   */
+  public static final String ENV_VAR_NEW_RELIC_METRIC_PATH = "NEW_RELIC_METRIC_PATH";
+
+  /**
+   * The name of the environment variable that should be used to provide the {@link
+   * #getMetricOptions()} {@link MetricOptions#getNewRelicMetricPeriod()} value.
+   */
+  public static final String ENV_VAR_NEW_RELIC_METRIC_PERIOD = "NEW_RELIC_METRIC_PERIOD";
+
+  /**
+   * The name of the environment variable that should be used to provide the location of the flyway
+   * scripts.
+   */
+  public static final String ENV_VAR_FLYWAY_SCRIPT_LOCATION = "FLYWAY_SCRIPT_LOCATION";
+
+  private final MetricOptions metricOptions;
+  private final DatabaseOptions databaseOptions;
+  private final String flywayScriptLocationOverride;
```

Comment:
Indeed it is styled as an override; in normal operation we will use flyway's default location of src/main/resources/db/migration. I will document this, however, since that wasn't clear to me either.

(For fields we would likely not have documentation requirements, unless needed to explain something like this.)

Reply:
A discussion for a future time but I often find the field name javadocs to be some of the most helpful when trying to understand a class so I'm probably going to advocate that we include them in the mandatory javadocs when we get to that point.